pkg/agent/memberlist/cluster_test.go

```diff
@@ -0,0 +1,741 @@
+//BenchmarkCluster_ShouldSelect/select_node_from_1000_alive_nodes-16         	12263878	        95.5 ns/op
+//BenchmarkCluster_ShouldSelect/select_node_from_100_alive_nodes
+//BenchmarkCluster_ShouldSelect/select_node_from_100_alive_nodes-16          	13036746	       103 ns/op
+//BenchmarkCluster_ShouldSelect/select_node_from_10_alive_nodes
+//BenchmarkCluster_ShouldSelect/select_node_from_10_alive_nodes-16           	14923483	        77.5 ns/op
+func BenchmarkCluster_ShouldSelect(b *testing.B) {
+	genNodes := func(n int) []string {
+		nodes := make([]string, n)
+		for i := 0; i < n; i++ {
+			nodes[i] = fmt.Sprintf("node-%d", i)
+		}
+		return nodes
+	}
+	nodeSelectedForEgress := func(consistentHash *consistenthash.Map, egressName, nodeName string) bool {
+		return consistentHash.Get(egressName) == nodeName
+	}
+
+	benchmarkCases := []struct {
+		name       string
+		nodes      []string
+		egressName string
+		localNode  string
+	}{
+		{
+			name:       "select Node from 10000 alive nodes",
+			nodes:      genNodes(10000),
+			egressName: "egress-10",
+			localNode:  "node-10",
+		},
+		{
+			name:       "select Node from 1000 alive nodes",
+			nodes:      genNodes(1024),
+			egressName: "egress-10",
+			localNode:  "node-10",
+		},
+		{
+			name:       "select Node from 100 alive nodes",
+			nodes:      genNodes(128),
+			egressName: "egress-10",
+			localNode:  "node-10",
+		},
+		{
+			name:       "select Node from 10 alive nodes",
+			nodes:      genNodes(8),
+			egressName: "egress-10",
+			localNode:  "node-10",
+		},
+	}
+
+	for i := range benchmarkCases {
+		bc := benchmarkCases[i]
+		b.Run(fmt.Sprintf("%s-nodeSelectedForEgress", bc.name), func(b *testing.B) {
+			b.ResetTimer()
+			consistentHash := consistenthash.New(defaultVirtualNodeReplicas, nil)
+			consistentHash.Add(bc.nodes...)
+			for i := 0; i < b.N; i++ {
+				nodeSelectedForEgress(consistentHash, bc.egressName, bc.localNode)
+			}
+		})
+	}
+}
+
+func genRandomStr(num int) string {
+	buf := make([]byte, num)
+	_, err := rand.Read(buf)
+	if err != nil {
+		return ""
+	}
+	return fmt.Sprintf("%016x", buf)
+}
+
+func TestCluster_ShouldSelect(t *testing.T) {
+	nodes := []string{"node1"}
+
+	genLocalNodeCluster := func(nodeNme string) *Cluster {
+		cluster := &Cluster{
+			mList:             &memberlist.Memberlist{},
+			nodeName:          nodeNme,
+			consistentHashMap: make(map[string]*consistenthash.Map),
+		}
+		return cluster
+	}
+
+	node1Cluster := genLocalNodeCluster("node1")
+	node1Cluster.consistentHashMap["default"] = consistenthash.New(defaultVirtualNodeReplicas, nil)
+	node1Cluster.consistentHashMap["default"].Add(nodes...)
+
+	egressNum := 3
+	hitCount := 0
+	for i := 0; i < egressNum; i++ {
+		egressName := fmt.Sprintf("%s-%d", genRandomStr(10), i)
+		if node1Cluster.consistentHashMap["default"].Get(egressName) == nodes[0] {
+			hitCount++
+		}
+	}
+	assert.Equal(t, egressNum, hitCount, "Egress total num should be equal")
```

Comment:
Ditto, I think this one can just be removed if you add the test I suggested above.

Reply:
done